static/src/js/components/Map/Map.jsx

```diff
@@ -286,6 +330,59 @@ const Map = ({
   const mapRef = useRef()
   const mapElRef = useRef()
 
+  useEffect(() => {
+    // Set initial base layer visibility
+    worldImageryLayer.setVisible(base.worldImagery)
+    correctedReflectanceLayer.setVisible(base.trueColor)
+    landWaterMapLayer.setVisible(base.landWaterMap)
+
+    // Set initial overlay layer visibility
+    bordersRoadsLayer.setVisible(overlays.referenceFeatures)
+    coastlinesLayer.setVisible(overlays.coastlines)
+    placeLabelsLayer.setVisible(overlays.referenceLabels)
+  }, [])
+
+  const handleLayerChange = ({ id, checked }) => {
+    // Handle base layers
+    if (baseLayerIds.includes(id)) {
+      const newBase = {
+        worldImagery: false,
+        trueColor: false,
+        landWaterMap: false
+      }
+
+      // Set the selected base layer to true in the Redux state
+      if (id === mapLayers.worldImagery) newBase.worldImagery = checked
+      if (id === mapLayers.correctedReflectance) newBase.trueColor = checked
+      if (id === mapLayers.landWaterMap) newBase.landWaterMap = checked
+
+      // Update all base layer visibility
+      Object.keys(baseLayers).forEach((layerId) => {
+        baseLayers[layerId].setVisible(layerId === id && checked)
+      })
+
+      onChangeMap({ base: newBase })
+    } else {
+      // Handle overlay layers
+      const newOverlays = { ...overlays }
+
+      // Update the overlay layer visibility if it exists in our mapping
+      if (id in overlayLayers) {
+        overlayLayers[id].setVisible(checked)
+
+        // Update the corresponding property in overlays state
+        // (maintaining the mapping between layer IDs and state property names)
+        if (id === mapLayers.bordersRoads) newOverlays.referenceFeatures = checked
```

Comment:
You can do this without so many conditionals, `newOverlays[id] = checked` does the same thing as these 3 conditional checks